### PR TITLE
fix for access array data with indices

### LINKED ISSUE #9068: Access Array Data with Indexes problem needs a little tweak
#### Challenge Name

Access Array Data with Indexes

https://www.freecodecamp.com/challenges/access-array-data-with-indexes#?solution=%0A%2F%2F%20Example%0Avar%20ourArray%20%3D%20%5B1%2C2%2C3%5D%3B%0Avar%20ourData%20%3D%20ourArray%5B0%5D%3B%20%2F%2F%20equals%201%0A%0A%2F%2F%20Setup%0Avar%20myArray%20%3D%20%5B1%2C2%2C3%5D%3B%0A%0A%2F%2F%20Only%20change%20code%20below%20this%20line.%0A%0A

Instructions
Create a variable called myData and set it to equal the first value of myArray.

#### Issue Description

It is not an issue rather we can get solution by not following steps

**here is the wrong solution which is accepted**

```
var ourArray = [1,2,3];
var ourData = ourArray[0]; // equals 1

// Setup
var myArray = [1,2,3];

// Only change code below this line.
var myData = 1;
```

**solution to be accepted**
```

var ourArray = [1,2,3];
var ourData = ourArray[0]; // equals 1

// Setup
var myArray = [1,2,3];

// Only change code below this line.
var myData = myArray[0];
```

### PR DESCRIPTION
**Pull request by ghost**
- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) them into one commit).
- [x] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change
- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)
#### Checklist:
- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #9068 
#### Description

Added assertion to check code for `\s*=\s*myArray[0]`
Added corresponding message 

> The data in variable `myArray` should be accessed using bracket notation

### CODE REVIEW

**Comment by robjloranger** [comment]
sorry, forgot to rename the branch

**Comment by atjonathan** [comment]
There seems to be some merge conflicts.
```bash
git checkout staging
git pull --rebase upstream staging
git checkout fix__using-array-indices
git rebase staging
```

**Comment by atjonathan** [comment]
@robjloranger, you also need to squash your commits!

```bash
git rebase -i HEAD~3
```

> Change the 2nd and 3rd commits from `pick` to `squash`.

```bash
git push -f origin fix__using-array-indices
```

**Comment by robjloranger** [comment]
@atjonathan i'm having quite the day lol
it wants to include someone else commit..
pick b730761 This is a buggy challange (#8843)
and is not including my latest commit

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 1]
```diff
@@ -1693,7 +1693,8 @@
         "var myArray = [1,2,3];\nvar myData = myArray[0];"
       ],
       "tests": [
-        "assert((function(){if(typeof myArray != 'undefined' && typeof myData != 'undefined' && myArray[0] === myData){return true;}else{return false;}})(), 'message: The variable <code>myData</code> should equal the first value of <code>myArray</code>.');"
+        "assert((function(){if(typeof myArray != 'undefined' && typeof myData != 'undefined' && myArray[0] === myData){return true;}else{return false;}})(), 'message: The variable <code>myData</code> should equal the first value of <code>myArray</code>.');",
```
Comment (crashuniverse) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: Please use `!==` type checking inequality operator. `typeof myArray != 'undefined'` => `typeof myArray !== 'undefined'`

**Comment by crashuniverse** [comment]
In the issue description you use `[x ]` while you should remove that space and make it `[x]`. It makes them checkboxes and you can see it in Preview tab before updating the comment.

File: seed/challenges/01-front-end-development-certification/basic-javascript.json (hunk 1 above)
Comment (robjloranger) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: That's not my code, I can change it if you like though.

Comment (crashuniverse) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: Yeah, no one will probably ever get closer to this line than you.

**Comment by crashuniverse** [comment]
LGTM

**Comment by atjonathan** [comment]
#### **LGTM :+1:**
Thanks for contributing :tada:

**Pull request merged**